cogeo_mosaic/backends/base.py

```diff
@@ -18,18 +19,57 @@
 
 from cogeo_mosaic.backends.utils import find_quadkeys, get_hash
 from cogeo_mosaic.cache import cache_config
-from cogeo_mosaic.errors import NoAssetFoundError
+from cogeo_mosaic.errors import NoAssetFoundError, UnsupportedOperation
 from cogeo_mosaic.models import Info, Metadata
 from cogeo_mosaic.mosaic import MosaicJSON
 from cogeo_mosaic.utils import bbox_union
 
 
+def update_zoom_and_bounds(instance, attribute, value):
+    """Update Zooms and Bounds attribute when mosaic_def is set."""
+    instance.minzoom = value.minzoom
+    instance.maxzoom = value.maxzoom
+    instance.bounds = value.bounds
+    return value
+
+
+def mode_validator(instance, attribute, value):
+    """Validate mode."""
+    if value not in instance._available_modes:
+        raise ValueError(
+            f"Invalid '{value}' mode. MUST be one of {instance._available_modes}."
+        )
+
+    if "r" in value:
+        instance._readable = True
+
+    elif "w" in value:
+        instance._writable = True
+
+    if "+" in value:
+        instance._writable = True
+
+
 @attr.s
 class BaseBackend(BaseReader):
```

Comment:
Do we still use the fact that it subclasses from `BaseReader`? There's a separate `reader` argument to the class.

Reply:
it has to be a subclass of rio_tilie.io.BaseReader to be compatible with other project (e.g TiTiler). The reader argument just tells the mosaic backend what reader to use for the assets